libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1911,8 +1911,47 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			break;
 		}
 		case Type::Category::Struct:
-			solUnimplementedAssert(false, "Struct conversion not implemented.");
+		{
+			solAssert(toCategory == Type::Category::Struct, "");
```

Comment:
The implementation in the old code generator literally is just four lines:
```
				m_context << Instruction::DUP1;
				m_context << Instruction::CALLDATASIZE;
				m_context << Instruction::SUB;
				abiDecode({&targetType}, false);
```
Why does this need so much code?

Reply:
Usable part of code, all else is basically asserts.
```
let <memberParams> := <abiDecode>(value, add(value, <dataSize>))
converted := <allocStruct>()
<#member>
	mstore(add(converted, <memberOffset>), <memberValue>)
</member>
```

If there exists tupleDecoder which decodes from calldata to specific position in memory, it would be 2 lines of code...

Reply:
something like this:
```
converted := <allocStruct>()
<abiDecode>(converted, value, add(value, <dataSize>))
```

For loop and member params stuff could also be removed in that case.

Reply:
Why do you need to call `<allocStruct>()`? The tupleDecoder should do that for you. The comment says 'can allocate memory'.

Reply:
I was thinking that there is no tuple decoder on struct types and was wrong...
Fixed now, it looks very simple :)